libres/lib/enkf/enkf_main.cpp

```diff
@@ -538,10 +538,16 @@ int enkf_main_load_from_run_context(enkf_main_type *enkf_main,
 
                         state_map_update_undefined(state_map, realisation,
                                                    STATE_INITIALIZED);
-
-                        return enkf_state_load_from_forward_model(
-                            enkf_main_iget_state(enkf_main, realisation),
-                            ert_run_context_iget_arg(run_context, realisation));
+                        try {
+                            return enkf_state_load_from_forward_model(
+                                enkf_main_iget_state(enkf_main, realisation),
+                                ert_run_context_iget_arg(run_context,
+                                                         realisation));
+                        } catch (const std::invalid_argument) {
+                            state_map_iset(state_map, realisation,
+                                           STATE_LOAD_FAILURE);
+                            return 1;
```

Comment:
Does this make sense when the function actually returns how many was loaded?

Reply:
You are very correct! Was a bit puzzel when I had to return 1 instead of 0, but with the help of @dafeda we figured out that this should really return an enum instead, so temporarily: `return (int) LOAD_FAILURE;`, but will try to add another commit so it has the correct return type, if that turns out to be a lot of work I will create an issue.

Reply:
Replaced it with enums all around, should be more readable now.